Dockerfile

```diff
@@ -0,0 +1,47 @@
+ARG GO_VERSION=1.18
+ARG RUNC_VERSION=v1.1.3
+
+# syntax = docker/dockerfile:1.4.0
+FROM --platform=${BUILDPLATFORM} golang:${GO_VERSION}-alpine AS base
+WORKDIR /src
+ENV CGO_ENABLED=0
+COPY go.* .
+# https://go.dev/ref/mod#module-cache
+RUN --mount=type=cache,target=/go/pkg/mod go mod download
+
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.1.1@sha256:23ca08d120366b31d1d7fad29283181f063b0b43879e1f93c045ca5b548868e9 AS xx
+
+FROM base AS build
+
+COPY --from=xx / /
+
+ARG TARGETPLATFORM
+# https://pkg.go.dev/cmd/go#hdr-Build_and_test_caching
+RUN --mount=type=bind,target=. \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+  xx-go build -o /out/buildg .
+
+FROM --platform=${BUILDPLATFORM} golang:${GO_VERSION}-bullseye AS build-base-debian
+RUN dpkg --add-architecture arm64 && \
+  dpkg --add-architecture amd64 && \
+  apt-get update && \
+  apt-get install -y crossbuild-essential-amd64 crossbuild-essential-arm64 git libbtrfs-dev:amd64 libbtrfs-dev:arm64 libseccomp-dev:amd64 libseccomp-dev:arm64
+
+FROM build-base-debian AS build-runc
+ARG {RUNC_VERSION}
+RUN git clone https://github.com/opencontainers/runc.git /go/src/github.com/opencontainers/runc
+WORKDIR /go/src/github.com/opencontainers/runc
+RUN git checkout ${RUNC_VERSION} && \
+  mkdir -p /out
+ENV CGO_ENABLED=1
+RUN GOARCH=amd64 CC=x86_64-linux-gnu-gcc make static && \
+  cp -a runc /out/runc.amd64
+RUN GOARCH=arm64 CC=aarch64-linux-gnu-gcc make static && \
+  cp -a runc /out/runc.arm64
+
+FROM ubuntu:bionic AS bin-unix
+COPY --from=build /out/buildg /
+COPY --from=build-runc /out/runc.${TARGETARCH:-amd64} /usr/local/bin/runc
```

Comment:
Can we use `20.04` which is well tested in our GithubActions CI ?
And, could you add `ca-certificates` apt pkg for allowing buildg to access DockerHub etc. from the container.
Currently it fails.

```console
$ docker run --rm -it --privileged -v /tmp/ctx:/ctx test debug /ctx/
WARN[2022-07-27T08:30:03Z] using host network as the default            
WARN[2022-07-27T08:30:03Z] git source cannot be enabled: failed to find git binary: exec: "git": executable file not found in $PATH 
#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile:
#1 transferring dockerfile: 237B done
#1 DONE 0.2s

#2 [internal] load .dockerignore
#2 transferring context: 2B done
#2 DONE 0.2s

#3 [internal] load metadata for docker.io/library/busybox:latest
INFO[2022-07-27T08:30:04Z] trying next host                              error="failed to do request: Head \"https://registry-1.docker.io/v2/library/busybox/manifests/latest\": x509: certificate signed by unknown authority" host=registry-1.docker.io
#3 ERROR: failed to do request: Head "https://registry-1.docker.io/v2/library/busybox/manifests/latest": x509: certificate signed by unknown authority
------
 > [internal] load metadata for docker.io/library/busybox:latest:
------
failed to build: failed to solve: failed to do request: Head "https://registry-1.docker.io/v2/library/busybox/manifests/latest": x509: certificate signed by unknown authority
```

Reply:
Of course 🙋🏻‍♂️